### PR TITLE
Quill tests

### PR DESCRIPTION
**Pull request by gabro**
Still WIP

### CODE REVIEW

**Comment by gabro** [comment]
@bytecodeguru the failure is due to Slick Tests, but I think #33 will fix them. I propose we merge this as is and then address #33

File: backend/quill/src/main/scala/io.buildo.toctoc/quill/authentication/token/QuillAccessTokenAuthenticationDomain.scala [hunk 1]
```diff
@@ -34,7 +35,7 @@ class QuillAccessTokenAuthenticationDomain[D <: SqlIdiom, N <: NamingStrategy, C
   private implicit val instantEncoder: Encoder[Instant] =
     encoder[Instant](SqlTypes.TIMESTAMP_WITH_TIMEZONE)
   private implicit val instantDecoder: Decoder[Instant] =
-    decoder[Instant]({ case i: Instant => i }, SqlTypes.TIMESTAMP_WITH_TIMEZONE)
+    decoder[Instant]({ case d: JodaDateTime => d.toDate.toInstant }, SqlTypes.TIMESTAMP_WITH_TIMEZONE)
```
Comment (bytecodeguru) [inline backend/quill/src/main/scala/io.buildo.toctoc/quill/authentication/token/QuillAccessTokenAuthenticationDomain.scala]: I don't understand what is happening here, why Joda?

File: backend/quill/src/test/scala/io.buildo.toctoc/quill/authentication/test/QuillLoginAuthenticationDomainFlowSpec.scala [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    val token3 = authFlow.exchangeForTokens(login2).futureValue.right.value
+    authFlow.unregisterAllSubjectTokens(subject).futureValue
+    authFlow.validateToken(token).futureValue shouldBe 'left
+    authFlow.validateToken(token2).futureValue shouldBe 'left
+    authFlow.validateToken(token3).futureValue shouldBe 'right
+  }
+
+  "single subject credentials unregistration" should "take effect" in {
+    authFlow.unregisterLogin(login).futureValue
+    authFlow.exchangeForTokens(login).futureValue shouldBe 'left
```
Comment (bytecodeguru) [inline backend/quill/src/test/scala/io.buildo.toctoc/quill/authentication/test/QuillLoginAuthenticationDomainFlowSpec.scala]: Perhaps this test is missing a precondition:

```
authFlow.registerSubjectLogin(subject, login).futureValue shouldBe 'right
```

File: backend/slick/src/main/scala/io.buildo.toctoc/slick/authentication/login/SlickLoginAuthenticationDomain.scala [hunk 3]
```diff
@@ -25,7 +25,7 @@ class SlickLoginAuthenticationDomain(db: Database)(implicit ec: ExecutionContext
 
     def * = (id, ref, username, passwordHash)
 
-    def uniqueRefIdx = index("unique_ref_idx", username, unique = true)
+    def uniqueUsernameIdx = index("unique_username_idx", username, unique = true)
```
[the remark this answers is not in the document]
Comment (bytecodeguru) [reply backend/slick/src/main/scala/io.buildo.toctoc/slick/authentication/login/SlickLoginAuthenticationDomain.scala]: 💣

File: backend/quill/src/main/scala/io.buildo.toctoc/quill/authentication/token/QuillAccessTokenAuthenticationDomain.scala (hunk 1 above)
Comment (gabro) [reply backend/quill/src/main/scala/io.buildo.toctoc/quill/authentication/token/QuillAccessTokenAuthenticationDomain.scala]: Joda is used internally by quill, so we're given a `JodaDateTime`

File: backend/quill/src/test/scala/io.buildo.toctoc/quill/authentication/test/QuillLoginAuthenticationDomainFlowSpec.scala (hunk 2 above)
Comment (gabro) [reply backend/quill/src/test/scala/io.buildo.toctoc/quill/authentication/test/QuillLoginAuthenticationDomainFlowSpec.scala]: 👍 makes sense, I'll add it

**Pull request merged**